main/src/ca/uwaterloo/flix/language/phase/jvm/GenExpression.scala

```diff
@@ -859,20 +859,37 @@ object GenExpression {
     case Expression.Spawn(exp1, exp2, _, loc) =>
       addSourceLine(visitor, loc)
 
-      // Compile the expression, putting a function implementing the Runnable interface on the stack
-      compileExpression(exp1, visitor, currentClass, lenv0, entryPoint)
-      visitor.visitTypeInsn(CHECKCAST, JvmName.Runnable.toInternalName)
-
-      // make a thread and run it
-      if (flix.options.xvirtualthreads) {
-        visitor.visitMethodInsn(INVOKESTATIC, "java/lang/Thread", "startVirtualThread", s"(${JvmName.Runnable.toDescriptor})${JvmName.Thread.toDescriptor}", false)
-        visitor.visitInsn(POP)
-      } else {
-        visitor.visitTypeInsn(NEW, "java/lang/Thread")
-        visitor.visitInsn(DUP_X1)
-        visitor.visitInsn(SWAP)
-        visitor.visitMethodInsn(INVOKESPECIAL, "java/lang/Thread", "<init>", s"(${JvmName.Runnable.toDescriptor})${JvmType.Void.toDescriptor}", false)
-        visitor.visitMethodInsn(INVOKEVIRTUAL, "java/lang/Thread", "start", AsmOps.getMethodDescriptor(Nil, JvmType.Void), false)
+      exp2 match {
+        // The expression represents the `Static` region, just start a thread directly
```

Comment:
This assumes `Region = Static`, right?

Reply:
Correct (which is true, I believe)

Reply:
And so once `Static` is gone, `Region` would be gone too?